platform/engineering/design-docs/2022-03-11-schedule-pending-appt.md

```diff
@@ -0,0 +1,170 @@
+# Scheduling Pending Appointments
+
+**Author(s):** Andrew Herzberg  
+**Last Updated:** March 11,2022  
+**Status:** **Draft** | In Review | Approved  
+**Approvers:** _Person A_ \[ \], _Person B_ \[ \], ...  
+
+## Overview
+
+### Objective
+The goal is to add the feature to the mobile app to schedule pending appointments. This functionality is currently available in `vets-website` so much of the logic can be utilized in the mobile app.
+
+The intended audience is for the engineering team.
+
+### Background
+Currently, the `vets-website` scheduling feature is as follows:
+
+![XL9BReCm4DrpYhb0Ue0igaIehRfgrQZT4tYGQOrDFKCaRbz3BOXYb6mGiZtly-RjH0klVMkoJu6dpICTuC_mMyfTs_QMTQ2yy11wCjpmaSruye7V7g94cNKACgXrIs_M35IUKFu8rSw7NoUmf4C7SZL9mEKIsJy-RG9LeBnB99spWkmq4RbnSX8QnaxE2OI-zXJ8yfoudhv2c00C4alFHJd2CeHURP3O5ihEhhgP3-jbflQk](https://user-images.githubusercontent.com/1910447/158439763-fd97627e-1bb7-4053-bb30-86436b64a2a4.png)
+
+- Pick type of care to schedule an appointment for
+   - Check community care eligibility  
+   - `/vaos/v0/community_care/eligibility/{type of care}`
+   - Receive true or false for type of care
+- If type of care is eligible, user can select to make appointment request at VA facility or community cares facility
+- User is given list of facilities to choose from
+  - Get facility data 
+  - `/vaos/v2/facilities?children=true&ids[]={facility id}&ids[]={facility id}`
+  - Note: facility id's come from registered facilities in `/v0/user`
+  - Get additional facility scheduling information such as services provided and if available to schedule request (including child facilities)
+  - `/vaos/v2/scheduling/configurations?facility_ids[]={facility_id}&facility_ids[]={facility_id}`
+- User chooses facility
+  - Check if able to schedule request at given facility and service 
+  - `/vaos/v2/eligibility?facility_id={id}&clinical_service_id={service_type}&type=request`
+  - Multiple checks and responses are done here:
```

Comment:
So it looks like this is where those user specific checks occur. So maybe it is possible to cache the above results and then the only external requests we have to make for eligibility are for user data and this.

Reply:
yes, that'd be a good way of caching it.